tests/testthat/test-carehomes-check.R

```diff
@@ -48,6 +48,38 @@ test_that("everyone is infected when beta is large", {
 })
 
 
+test_that("noone stays in R, R_neg or PCR_neg if waning rate is very large", {
+  p <- carehomes_parameters(0, "england", beta_value = 0,
+                            waning_rate = 1e9)
+  mod <- carehomes$new(p, 0, 1)
+  info <- mod$info()
+
+  state <- carehomes_initial(info, 1, p)$state
+
+  # move everyone to R and R_neg initially
+  index_S <- array(info$index$S, info$dim$S)
+  index_R <- array(info$index$R, info$dim$R)
+  index_R_neg <- array(info$index$R_neg, info$dim$R_neg)
+  index_PCR_neg <- array(info$index$PCR_neg, info$dim$PCR_neg)
+  state[index_R] <- state[index_S]
+  state[index_R_neg] <- state[index_S]
+  state[index_PCR_neg] <- state[index_S]
+  state[index_S] <- 0
+
+  mod$set_state(state)
+  mod$set_index(integer(0))
+  y <- mod$transform_variables(drop(
+    dust::dust_iterate(mod, seq(0, 400, by = 4))))
+
+  # other than in the 4th age group (where infections are seeded)
+  # after the first time step, R is empty
+  expect_true(all(y$R[-4, -1] == 0))
```

Comment:
if only one event happens in any given timestep, surely people should spent one timestep in R?

Reply:
right - meant day, not timestep - there are 4 time steps per day as far as I can tell. 
Maybe @edknock you can confirm I understand correctly that
`dust::dust_iterate(mod, seq(0, 400, by = 4)`
runs the model mod for 100 days with time step 1/4 day?

Reply:
things should still transition through R though? if it's always empty that sounds like a problem to me

Reply:
so I initiate the model so everyone is in R instead of S (apart from age group 4 where the seeding is done so some people are in E or I and I didn't want to bother with that). In all other age groups everyone usually starts in S, but I've moved them all to R initially, then with a large waning rate they should all leave R immediately.

Reply:
I have tried to clarify in comments in the test, see [here](https://github.com/mrc-ide/sircovid/blob/ebcb2cda219dbf496258e553d898efb02243d3c8/tests/testthat/test-carehomes-check.R#L53).